kore/test/Test/Kore/Builtin/KEqual.hs

```diff
@@ -83,11 +83,12 @@ test_KEqual =
         actual <- evaluate original
         assertEqual' "" expect actual
     , testCaseWithoutSMT "kseq(x, dotk) equals kseq(x, dotk)" $ do
-        let expect = OrPattern.top
+        let expect = OrPattern.topOf kSort
             xConfigElemVarKItemSort =
                 configElementVariableFromId "x" kItemSort
             original =
-                mkEquals_
+                mkEquals
+                    kSort
```

Comment:
I was thinking about this, and I want to be 100% sure I understand this. I'd expect the `==_` (where `_` is `K`, `Int` etc.) to return a `Bool`. The reason this doesn't return `Bool` here but `kSort` is because `\\equals` is a predicate and we can give it any result sort. If we did `\\and` unification instead, the resulting sort would be `Bool`, right?

Reply:
Yes, if I replace `mkEquals` with `mkAnd`, the result sort is `Bool`

Reply:
Cool. Could you add some similar tests for `and` as well?